File: java/src/main/java/vsdk/toolkit/io/bin/reader/ScopedArrayBuffer.java
package vsdk.toolkit.io.bin.reader;

public class ScopedArrayBuffer<T> {
    private T value;

    public ScopedArrayBuffer() {
        this(null);
    }

    public ScopedArrayBuffer(T initialValue) {
        value = initialValue;
    }

    public void reset(T newValue) {
        value = newValue;
    }

    public T get() {
        return value;
    }
}

File: typescript/src/main/vsdk/toolkit/io/bin/reader/ScopedArrayBuffer.ts
export class ScopedArrayBuffer<T> {
  private value: T | null;

  public constructor(initialValue: T | null = null) {
    this.value = initialValue;
  }

  public reset(newValue: T | null): void {
    this.value = newValue;
  }

  public get(): T | null {
    return this.value;
  }
}
